File: OrderInventory-Backend/src/main/java/com/orderinventory/services/InventoryService.java
package com.orderinventory.services;

import java.util.List;

import com.orderinventory.dto.CustomInventoryDto;
import com.orderinventory.dto.InventoryDetailStoreIdDto;
import com.orderinventory.dto.InventoryDetailsDto;
import com.orderinventory.dto.InventoryDto;
import com.orderinventory.dto.InventoryWithShipmentsDto;
import com.orderinventory.dto.ShipmentStatusCountDTO;

public interface InventoryService {
	List<InventoryDto> getAllInventory();

	List<InventoryDetailStoreIdDto> getInventoryDetailsByStoreId(Integer storeId);

	List<InventoryWithShipmentsDto> getInventoryWithShipmentsByDeliveryAddress(String deliveryAddress);

	CustomInventoryDto getInventoryByOrderId(Integer orderId);

	List<ShipmentStatusCountDTO> countProductsByShipmentStatus(String shipmentStatus);

	List<InventoryDetailsDto> getInventoryDetailsByOrderId(Integer orderId);

}

File: OrderInventory-Frontend/src/app/services/inventory.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, OnInit } from '@angular/core';
import { map, Observable } from 'rxjs';
 import {InventoryByOrderId, InventoryDetails, InventoryDetailStoreIdDto, InventoryResponse, InventoryWithShipmentsDto, ShipmentStatusCountDTO} from '../Interface/InventoryResponse';





@Injectable({
  providedIn: 'root'
})
export class InventoryService{

  
  constructor(private http: HttpClient) {}



  private baseUrl = 'http://localhost:8081/api/v1';

  // methods
 
  getAllInventory(): Observable<InventoryResponse[]> {
    return this.http.get<InventoryResponse[]>(`${this.baseUrl}/inventory`).pipe(
      map((response: InventoryResponse[]) => {
        return response.map(inventory => ({
          inventoryId: inventory.inventoryId,
          storeData: {
            storeId: inventory.storeData.storeId,

          },
          productData: {
            productId: inventory.productData.productId
          },
          productInventory: inventory.productInventory
        }));
      })
    );
  }


  getInventoryByStoreId(storeId: number): Observable<InventoryDetailStoreIdDto[]> {
    return this.http.get<InventoryDetailStoreIdDto[]>(`${this.baseUrl}/inventory/store/${storeId}`);
  }



  getInventoryWithShipmentsByDeliveryAddress(deliveryAddress: string): Observable<InventoryWithShipmentsDto[]> {
    const url: string = `${this.baseUrl}/inventory/shipment?deliveryAddress=${deliveryAddress}`;
    return this.http.get<InventoryWithShipmentsDto[]>(url);
  }




  getInventoryByOrderId(orderId: number): Observable<InventoryByOrderId> {
    return this.http.get<InventoryByOrderId>(`${this.baseUrl}/inventory/${orderId}`);
  }


  countProductsByShipmentStatus(shipmentStatus: string): Observable<ShipmentStatusCountDTO[]> {
    return this.http.get<ShipmentStatusCountDTO[]>(`${this.baseUrl}/shipment/count?shipmentStatus=${shipmentStatus}`);
  }


  getInventoryDetailsByOrderId(orderId: number): Observable<InventoryDetails[]> {
    const url = `${this.baseUrl}/inventory/${orderId}/details`;
    return this.http.get<InventoryDetails[]>(url);
  }


}
